### PR TITLE
Cypress/E2E: Automate test community listed test cases

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Issues Automated:
  - [MM-T423](https://github.com/mattermost/mattermost-server/issues/18177) Online Status - Statuses update in center, in member icon drop-down, and in DM LHS sidebar
  - [MM-T244](https://github.com/mattermost/mattermost-server/issues/18164) Webapp: Post menu item Mark as Unread appearance
  - [MM-T1615](https://github.com/mattermost/mattermost-server/issues/18161) Pasting code block text into external apps does not include line numbers
  - [MM-T257](https://github.com/mattermost/mattermost-server/issues/18168) Mark as Unread when bringing window into focus
  - [MM-T669](https://github.com/mattermost/mattermost-server/issues/15697) Custom slash command in DM and GM
  - [MM-T2367](https://github.com/mattermost/mattermost-server/issues/18174) Compact view

### CODE REVIEW

**Comment by Phrynobatrachus** [comment]
/e2e-test INCLUDE_FILE="custom_slash*, slash_commands*, mark_as*, copy_post*, check_user*, channel_scroll*"

File: e2e/cypress/tests/integration/scroll/channel_scroll_spec.js [hunk 1]
```diff
@@ -76,6 +75,37 @@ describe('Scroll', () => {
...
+        });
+
+        // # Set display mode to compact
+        cy.apiSaveMessageDisplayPreference('compact');
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        // * Verify that messages have user's name in post
+        cy.getLastPostId().then((parentMessageId) => {
+            cy.get(`#${parentMessageId}_message`).parent().invoke('text').then((text) => {
+                expect(text).to.equal(`sysadmin: ${message}`);
```
Comment (Phrynobatrachus) [inline e2e/cypress/tests/integration/scroll/channel_scroll_spec.js:104]: The `Update your status` button seems to be causing issues on this assertion for me.

```suggestion
                expect(text).to.contain('sysadmin');
                expect(text).to.contain(message);
```

**Review by Phrynobatrachus — CHANGES_REQUESTED** [review]
I noticed one small issue locally but otherwise looks great, thanks!

File: e2e/cypress/tests/integration/scroll/channel_scroll_spec.js (hunk 1 above)
Comment (yasserfaraazkhan) [reply e2e/cypress/tests/integration/scroll/channel_scroll_spec.js]: Thank you @Phrynobatrachus .
I have made the suggested changes.

**Review by nidak7 — APPROVED** [review]

**Review by Phrynobatrachus — APPROVED** [review]

File: e2e/cypress/tests/integration/mark_as_unread/mark_as_unread_spec.js [hunk 2]
```diff
@@ -99,6 +99,29 @@ describe('Mark as Unread', () => {
...
 
+    it('MM-T257 Mark as Unread when bringing window into focus', () => {
+        // * Verify channels are unread
+        cy.get(`#sidebarItem_${channelA.name}`).should(beUnread);
+        cy.get(`#sidebarItem_${channelB.name}`).should(beUnread);
+
+        // # Nagigate to integration screeen (away from chat/main screen)
+        cy.visit(`/${team1.name}/integrations/`);
+
+        // # Nagigate back to chat/main screen
```
Comment (josephbaylon) [inline e2e/cypress/tests/integration/mark_as_unread/mark_as_unread_spec.js:110]:
```suggestion
        // # Navigate back to chat/main screen
```

File: e2e/cypress/tests/integration/integrations/custom_slash_commands/custom_slash_commands_spec.js [hunk 3]
```diff
@@ -41,13 +43,80 @@ describe('Slash commands', () => {
...
+
+            // # Run the command in a GM
+            cy.visit(`/${team1.name}/messages/@${user2.username}`);
+            cy.get('#post_textbox', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').clear().type(`/${dmTrigger}{enter}{enter}`);
+            cy.wait(TIMEOUTS.TWO_SEC);
+
+            // * Verify that the message was sent to the GM
+            cy.getLastPostId().then((postId) => {
+                cy.get(`#post_${postId}`).get('.Badge').contains('BOT');
+            });
```
Comment (josephbaylon) [inline e2e/cypress/tests/integration/integrations/custom_slash_commands/custom_slash_commands_spec.js:115]: Same as above for `runSlashCommand`

**Review by josephbaylon — CHANGES_REQUESTED** [review]
Thanks @yasserfaraazkhan !!
Looking good! Just a few minor suggestions below.
Also, I noticed there were several tests that were disabled. Are those on purpose?

**Comment by yasserfaraazkhan** [comment]
@josephbaylon Thank you!!  🙌

I've pushed the suggested changes.

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks for applying the change request!

**Comment by furqanmlk** [comment]
Thanks @yasserfaraazkhan 
Great Job 👍

**Review by furqanmlk — APPROVED** [review]

**Comment by saturninoabril** [comment]
/e2e-test INCLUDE_FILE="slash_commands_spec.js,mark_as_unread_spec.js,copy_post_text_spec.js,check_user_status_spec.js,channel_scroll_spec.js"

**Comment by saturninoabril** [comment]
/e2e-test INCLUDE_FILE="slash_commands_spec.js,mark_as_unread_spec.js,copy_post_text_spec.js,check_user_status_spec.js,channel_scroll_spec.js"

**Pull request merged**